server/src/main/java/com/objectcomputing/checkins/services/role/RoleController.java

```diff
@@ -91,9 +133,12 @@ public Role readRole(UUID id) {
      * @return {@link List < Role > list of roles}
      */
     @Get("/{?role,memberid}")
-    public Set<Role> findRole(@Nullable RoleType role,
-                              @Nullable UUID memberid) {
-        return roleServices.findByFields(role, memberid);
+    public Single<HttpResponse<Set<Role>>> findRole(@Nullable RoleType role,
+    @Nullable UUID memberid) {
+        return Single.fromCallable(() -> roleServices.findByFields(role, memberid))
+                .observeOn(Schedulers.from(eventLoopGroup))
+                .map(createdRole -> (HttpResponse<Set<Role>>) HttpResponse.ok(createdRole))
```

Comment:
same here

Reply:
updated to userRole for better readability